Studio crashing on edit draft directly from dashboard

**Describe the bug**
Getting a crash in studio when clicking on docs from the dashboard document list that are in `draft` and were never published.  I am uploading docs through the api with `drafts.` prepended to the id so they stay as drafts and not published and this is working fine if you go through the normal flow of the document panes but if you click from the dashboard it crashes with error `editOpsOf does not expect a draft id`

**To Reproduce**

Steps to reproduce the behavior:

1. Go to 'dashboard'
2. Click on 'latest doc that was created as a draft'
3. Studio attempts to load the edit draft like `http://localhost:3333/intent/edit/type=student;id=drafts.1221/`
4. See error in screenshot `react_devtools_backend.js:6 Error: editOpsOf does not expect a draft id.`

**Expected behavior**

It should load the doc in edit mode like other previously published documents.  NOTE this works fine when the document has been published before.  Once I publish it this works again. To trigger this it needs to be a newly created doc as a draft before publishing.

**Screenshots**
If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/817344/80761961-6fc22500-8af0-11ea-971c-d72d5e945b98.png)

**Which versions of Sanity are you using?**

Run `sanity versions` in the terminal and copy-paste the result here.
```
@sanity/cli                      1.149.9 (up to date)
@sanity/base                    1.149.10 (up to date)
@sanity/block-content-to-react     2.0.7 (up to date)
@sanity/cli                      1.149.9 (up to date)
@sanity/color-input              1.149.7 (up to date)
@sanity/components              1.149.10 (up to date)
@sanity/core                    1.149.12 (up to date)
@sanity/dashboard               1.149.10 (up to date)
@sanity/default-layout          1.149.10 (up to date)
@sanity/default-login           1.149.11 (up to date)
@sanity/desk-tool               1.149.10 (up to date)
@sanity/production-preview       1.149.0 (up to date)
@sanity/vision                   1.149.0 (up to date)
```

**What operating system are you using?**

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.

--- a/packages/@sanity/desk-tool/src/components/IntentResolver.js
+++ b/packages/@sanity/desk-tool/src/components/IntentResolver.js
@@ -3,6 +3,7 @@ import React, {useEffect, useState} from 'react'
 import {of} from 'rxjs'
 import {map} from 'rxjs/operators'
 import client from 'part:@sanity/base/client'
+import {getPublishedId} from 'part:@sanity/base/util/draft-utils'
 import {useRouter} from 'part:@sanity/base/router'
 import Spinner from 'part:@sanity/components/loading/spinner'
 import {useStructure} from '../utils/resolvePanes'
@@ -13,7 +14,20 @@ import {getTemplateById} from '@sanity/base/initial-value-templates'
 
 const FALLBACK_ID = '__fallback__'
 
+function removeDraftPrefix(documentId) {
+  const publishedId = getPublishedId(documentId)
+  if (publishedId !== documentId) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Removed unexpected draft id in document link: All links to documents should have the `draft.`-prefix removed and something appears to have made an intent link to `%s`',
+      documentId
+    )
+  }
+  return publishedId
+}
+
 /**
+ *
  * This is a *very naive* implementation of an intent resolver:
  * - If type is missing from params, it'lltry to resolve from document
  * - It manually builds a pane segment path: "<typeName>;<documentId>"
@@ -73,7 +87,7 @@ function getNewRouterState({structure, documentType, params, payload, documentId
   const template = isTemplateCreate && getTemplateById(params.template)
   const type = (template && template.schemaType) || documentType
   const fallbackParameters = {type, template: params.template}
-  const newDocumentId = documentId === FALLBACK_ID ? UUID() : documentId
+  const newDocumentId = documentId === FALLBACK_ID ? UUID() : removeDraftPrefix(documentId)
 
   return terminatesInDocument
     ? paneSegments

--- a/packages/@sanity/desk-tool/src/tool/index.js
+++ b/packages/@sanity/desk-tool/src/tool/index.js
@@ -7,8 +7,8 @@ import Icon from 'part:@sanity/base/view-column-icon'
 import {route, useRouterState} from 'part:@sanity/base/router'
 import {parsePanesSegment, encodePanesSegment} from '../utils/parsePanesSegment'
 import IntentResolver from '../components/IntentResolver'
-import DeskTool from './DeskTool'
 import {EMPTY_PARAMS} from '../constants'
+import DeskTool from './DeskTool'
 
 function toState(pathSegment) {
   return parsePanesSegment(decodeURIComponent(pathSegment))

